R/editing.R

```diff
@@ -749,20 +750,69 @@ update_resource_map <- function(mn,
   me <- get_token_subject()
   set_rights_holder(mn, resource_map_pid, me)
 
+  # Get the old resource map so we can extract any statements we need out of it
+  # such as PROV statements
+  old_resource_map_path <- tempfile()
+  writeLines(rawToChar(dataone::getObject(mn, resource_map_pid)), old_resource_map_path)
+  statements <- parse_resource_map(old_resource_map_path)
+  statements <- filter_packaging_statements(statements)
+  if (is.data.frame(other_statements)) {
+    statements <- rbind(statements,
+                        other_statements)
+  }
+
+  prov_pids <- gsub("https://cn-stage-2.test.dataone.org/cn/v[0-9]/resolve/|https://cn.dataone.org/cn/v[0-9]/resolve/|https://cn-stage.test.dataone.org/cn/v[0-9]/resolve/",
+                    "",
+                    c(statements$subject, statements$object)) %>%
+    gsub("%3A", ":", .)
+  prov_pids <- prov_pids[-(grep("^http", prov_pids))] %>% # might need to catch other things besides URLs
+    unique(.)
+
   # Create the replacement resource map
   if (is.null(identifier)) {
     identifier <- paste0("resource_map_", new_uuid())
   }
 
-  new_rm_path <- generate_resource_map(metadata_pid = metadata_pid,
+  if (keep_prov == FALSE){
```

Comment:
I notice that the call to `generate_resource_map` shows up four times here, in separate conditional branches. While it totally works, a slightly tidier pattern is to use the conditionals to generate the appropriates values for each parameter of `generate_resource_map` and call it once after the conditional block. The benefits are simplifying the conditional, and reducing any pain that may come later when if/when `generate_resource_map` is refactored which would cause use to have to update four call sites instead of one.

This'd look something like:

```r
if (CONDITION) {
  other_statements = NULL
}

generate_resource_map(...) # All args, even NULL'd-out ones like `other_statements`
```